packages/core/src/sync-store/postgres/migrations.ts

```diff
@@ -455,6 +456,73 @@ const migrations: Record<string, Migration> = {
         .execute();
     },
   },
+  "2024_03_20_0_checkpoint_in_logs_table": {
+    async up(db: Kysely<any>) {
+      await db.schema
+        .alterTable("logs")
+        .addColumn("checkpoint", "numeric(75, 0)")
+        .execute();
+      await db.executeQuery(
+        sql`
+          WITH checkpoint_vals AS (
+            SELECT logs.id, blocks.timestamp, blocks."chainId", blocks.number, logs."transactionIndex", logs."logIndex"
+            FROM ponder_sync.logs logs
+            JOIN ponder_sync.blocks blocks ON logs."blockHash" = blocks.hash
+          )
+          UPDATE ponder_sync.logs
+          SET checkpoint=
+              (lpad(checkpoint_vals.timestamp::text, 10, '0') ||
+              lpad(checkpoint_vals."chainId"::text, 16, '0') ||
+              lpad(checkpoint_vals.number::text, 16, '0') ||
+              lpad(checkpoint_vals."transactionIndex"::text, 16, '0') ||
+              '5' ||
+              lpad(checkpoint_vals."logIndex"::text, 16, '0'))::numeric(75, 0)
+          FROM checkpoint_vals
+          WHERE ponder_sync.logs.id = checkpoint_vals.id;
+        `.compile(db),
+      );
+
+      // sanity check our checkpoint encoding on the first 10 rows of the table
+      const checkRes = await db.executeQuery(
```

Comment:
I might add the type here, eg
```ts
const checkRows = db.executeQuery<{
  timestamp: number;
  chainId: number;
  number: number;
  transactionIndex: number;
  logIndex: number;
  checkpoint: bigint;
}>( ...
```

And then below, you should be able to pass these inline:
```ts
        const expected = encodeCheckpoint({
          blockTimestamp: row.timestamp,
          chainId: row.chainId,
          blockNumber: row.number,
          // ... etc
        });
```